plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.googleapis.internal;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.appengine.v1.Appengine;
+import com.google.api.services.appengine.v1.Appengine.Apps;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager.Projects;
+import com.google.cloud.tools.eclipse.googleapis.GoogleApiException;
+import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
+import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.LoadingCache;
+import java.util.concurrent.ExecutionException;
+import org.eclipse.core.net.proxy.IProxyChangeEvent;
+import org.eclipse.core.net.proxy.IProxyChangeListener;
+import org.eclipse.core.net.proxy.IProxyService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+/**
+ * Class to obtain various Google cloud Platform related APIs.
+ */
+@Component
+public class GoogleApiFactory implements IGoogleApiFactory {
+
+  private IProxyService proxyService;
+
+  private JsonFactory jsonFactory;
+  private final ProxyFactory proxyFactory;
+  private LoadingCache<GoogleApiUrl, HttpTransport> transportCache;
+
+  private final IProxyChangeListener proxyChangeListener = new IProxyChangeListener() {
+    @Override
+    public void proxyInfoChanged(IProxyChangeEvent event) {
+      if (transportCache != null) {
+        transportCache.invalidateAll();
+      }
+    }
+  };
+
+  public GoogleApiFactory() {
+    this(new ProxyFactory());
+  }
+
+  @VisibleForTesting
+  public GoogleApiFactory(ProxyFactory proxyFactory) {
+    Preconditions.checkNotNull(proxyFactory, "proxyFactory is null");
+    this.proxyFactory = proxyFactory;
+  }
+
+  @Activate
+  public void init() {
+    jsonFactory = new JacksonFactory();
```

Comment:
Any reason you're not initializing this and making it _final_?  You set mocks in the tests, but don't seem to do anything with the mocks.